src/packet_analysis/Analyzer.cc

```diff
@@ -166,30 +166,67 @@ void Analyzer::Weird(const char* name, Packet* packet, const char* addl) const
 	session_mgr->Weird(name, packet, addl, GetAnalyzerName());
 	}
 
+void Analyzer::EnqueueAnalyzerConfirmationInfo(session::Session* session, zeek::Tag arg_tag)
+	{
+	static auto info_type = zeek::id::find_type<RecordType>("AnalyzerConfirmationInfo");
+	static auto info_c_idx = info_type->FieldOffset("c");
+
+	auto info = make_intrusive<RecordVal>(info_type);
+	if ( session )
+		info->Assign(info_c_idx, session->GetVal());
+
+	const auto& tval = arg_tag ? arg_tag.AsVal() : tag.AsVal();
+	event_mgr.Enqueue(analyzer_confirmation_info, tval, info);
+	}
+
+void Analyzer::EnqueueAnalyzerConfirmation(session::Session* session, zeek::Tag arg_tag)
+	{
+	// TBD: Should we just skip raising this event if session is NULL?
+	//      But also wondering what can be done with a confirmation that
+	//      doesn't have a session? Metrics tracking?
```

Comment:
Not immediately sure, but I'm realizing another issue here: raising a confirmation event per packet is pretty expensive. I'm wondering if confirmations just generally don't make sense for packet analyzers?

Reply:
Within Zeek the confirmation only happens once for a session using "AnalyzerState" (and from looking at the code only for tunnel analyzers that actually have an outer session), so that's not per-packet unless in worst-case tunneling scenario.

For violations as you pointed out below there's no cap per session, however, and these are likely more interesting.

> I'm wondering if confirmations just generally don't make sense for packet analyzers?

There is some basic DetectProtocol() logic, so it might make sense, but could also start tracking occurrences as members or metrics and make those available to script-land.

(...but I'm then tempted to question the generic events again if they are just for protocol and file analyzers...)

Reply:
> I'm wondering if confirmations just generally don't make sense for packet analyzers?

Minimally one effect would be that the `c$service` fields don't contain `ayiya` or `vxlan` for tunnel connections and we'd need to find a separate way to propagate those up.